zrml/rikiddo/src/types/ema_market_volume.rs

```diff
@@ -198,32 +218,72 @@ impl<F: FixedUnsigned + From<u32>> MarketAverage<F> for EmaMarketVolume<F> {
                 // against the timestamp of the previous transaction.
                 let timestamp_sub_start_time = volume.timestamp.saturating_sub(self.start_time);
 
+                let mut comparison_value = if let Some(res) = self.config.ema_period_estimate_after
+                {
+                    res.to_seconds()
+                } else {
+                    self.config.ema_period.to_seconds()
+                };
+
                 // It should not state transit, if the amount of gathered data is too low.
                 // This would result in a multiplier that is greater than 1, which can lead to
                 // a negative ema. The amount depends on the size of the smoothing factor.
-                if timestamp_sub_start_time > self.config.ema_period.to_seconds() as u64
+                if timestamp_sub_start_time > comparison_value as u64
                     && (*self.volumes_per_period() + 1.into()) >= self.config.smoothing
                 {
+                    // We extrapolate the txs per period
+                    if self.config.ema_period_estimate_after.is_some() {
+                        // Ensure that we don't divide by 0
+                        if comparison_value == 0 {
+                            comparison_value = 1
+                        }
+
+                        let premature_time_fixed = <FixedU128<U64>>::from(comparison_value);
+                        let mature_time_fixed =
+                            <FixedU128<U64>>::from(self.config.ema_period.to_seconds());
+                        // Overflow impossible
+                        let estimate_ratio = mature_time_fixed.saturating_div(premature_time_fixed);
+
+                        if estimate_ratio.int() > FU::max_value().to_num::<u64>() {
+                            // Cannot occur as long as the From<U32> trait is required for FU and
+                            // Timespan::to_seconds() returns u32.
+                            return Err("[EmaMarketVolume] Estimate ratio does not fit in FU");
+                        }
+
+                        // Can't panic due to the previous check
+                        let estimate_ratio_fu: FU = estimate_ratio.to_fixed();
+
+                        self.volumes_per_period = if let Some(res) =
+                            self.volumes_per_period.checked_mul(estimate_ratio_fu)
+                        {
+                            res.saturating_ceil()
+                        } else {
+                            return Err("[EmaMarketVolume] Overflow during estimation of \
+                                        transactions per period");
+                        }
+                    }
+
```

Comment:
This is an addition I added after some discussion with Numa about what should happen if the short period ema value is already present, but the long ema period still needs to gather data. One possible option, which you see implemented here, is that we simply extrapolate the transaction count for the long period based on the data that's available. For example, if we have a short period ema of 1 hour, we can configure the long period ema struct to collect 6 hours of data, but extrapolate the transaction count per period after 1 hour of collecting data. This way, the ema ratio `r = ema_short / ema_long` is already available after the short ema period and can be used to calculate the sigmoid fee, instead of using the initial fee.